site/app/controllers/forum/ForumController.php

```diff
@@ -496,7 +502,19 @@ public function publishPost() {
                 $result['next_page'] = $hasGoodAttachment[1];
             }
             else {
-                $post_id = $this->core->getQueries()->createPost($current_user_id, $post_content, $thread_id, $anon, 0, false, $hasGoodAttachment[0], $markdown, $parent_id);
+                $attachment_name = "";
+                if ($hasGoodAttachment[0] == 1) {
```

Comment:
It's almost always better to use PHP's [strict comparison](https://www.php.net/manual/en/language.operators.comparison.php) operators whenever possible.  Apply throughout.